### PR TITLE
Process resource stored message & run CSV detective

### PR DESCRIPTION
**Pull request by sixtedemaupeou**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+authors = [{name = "Sixte de Maupeou", email = "sixte.de-maupeou@data.gouv.fr"}]
+readme = "README.md"
+dynamic = ["version", "description"]
+dependencies = [
+    "boto3==1.21.21",
+    "click==8.0.4",
+    "kafka-python==2.0.2",
+    "redis==4.1.4",
+    "requests==2.27.1",
+    "celery==5.2.3"
```
Comment (geoffreyaldebert) [inline pyproject.toml:16]: Once published, we will need to set up csv-detective version here

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+boto3==1.21.21
+click==8.0.4
+flake8==4.0.1
+kafka-python==2.0.2
+pytest==7.1.1
+python-dotenv==0.19.2
+redis==4.1.4
+requests==2.27.1
+celery==5.2.3
```
Comment (geoffreyaldebert) [inline requirements.txt:9]: same thing here, we should add csv-detective once published.

File: udata_analysis_service/background_tasks.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    resource_id: str,
+    resource_location: dict,
+    minio_user: str,
+    minio_pwd: str,
+) -> None:
+    logging.info(
+        "Processing task for resource {} in dataset {}".format(resource_id, dataset_id)
+    )
+    analysis_report = routine(
+        minio_url=resource_location["netloc"],
```
Comment (geoffreyaldebert) [inline udata_analysis_service/background_tasks.py]: We mix here information relative to minio from .env and information delivered by kafka message. I think it is better if we keep all MINIO setup from env. 
We can above compare MINIO_URL from env with this `netloc` info and if it's not the same, do not process.

File: udata_analysis_service/background_tasks.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        minio_user=minio_user,
+        minio_pwd=minio_pwd,
+        minio_bucket=resource_location["bucket"],
```
Comment (geoffreyaldebert) [inline udata_analysis_service/background_tasks.py]: Same here, bucket for this service should be unique and be loaded from env. But we can compare info and do not process if it differs.

File: udata_analysis_service/background_tasks.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        minio_key=resource_location["key"],
```
Comment (geoffreyaldebert) [inline udata_analysis_service/background_tasks.py]: the key is to keep, though

File: udata_analysis_service/background_tasks.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        num_rows=50,
```
Comment (geoffreyaldebert) [inline udata_analysis_service/background_tasks.py]: If it is the number of rows analysed by csv-detective, I think it is a little bit low. Could we increase it to 500 and store it in an env variable ?

File: udata_analysis_service/background_tasks.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+        user_input_tests="ALL",
+        output_mode="LIMITED",
+        save_results=False,
```
Comment (geoffreyaldebert) [inline udata_analysis_service/background_tasks.py]: What does it means ?

Comment (geoffreyaldebert) [reply udata_analysis_service/background_tasks.py]: I would love to have a property save_result_to_minio and indicates where to store result report. But I think it is more a csv-detective feature.

**Review by geoffreyaldebert — APPROVED** [review]
🎊 Nice and clean. Approved if remarks are taken into account.
What I'm wondering is the size of this repo once a udata-kafka-core repo will be used. Code here will be minimal.

File: udata_analysis_service/background_tasks.py (hunk 7 above)
Comment (sixtedemaupeou) [reply udata_analysis_service/background_tasks.py]: save_results determines whether the results are saved locally as a JSON (useful for testing purposes) or not

Comment (sixtedemaupeou) [reply udata_analysis_service/background_tasks.py]: `upload_results` determines whether the results are uploaded to Minio but indeed the exact path is determined by `csv-detective`

File: udata_analysis_service/background_tasks.py (hunk 3 above)
Comment (sixtedemaupeou) [reply udata_analysis_service/background_tasks.py]: This is no longer the case, we now only use information coming from the message

File: udata_analysis_service/background_tasks.py (hunk 6 above)
Comment (sixtedemaupeou) [reply udata_analysis_service/background_tasks.py]: Done

**Review by geoffreyaldebert — APPROVED** [review]

**Comment by geoffreyaldebert** [comment]
🎉

**Pull request merged**